src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -223,5 +239,33 @@ public override Symbol AssociatedSymbol
                 return null;
             }
         }
+
+        internal override bool IsVirtualTupleField
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override sealed int GetHashCode()
+        {
+            return Hash.Combine(base.GetHashCode(), _name.GetHashCode());
+        }
+
+        public override sealed bool Equals(object obj)
+        {
+            return Equals(obj as TupleVirtualElementFieldSymbol);
+        }
+
+        public bool Equals(TupleVirtualElementFieldSymbol other)
```

Comment:
The base class `Equals(TupleFieldSymbol)` should be virtual and this should override that, otherwise two `TupleVirtualElementFieldSymbol`s with different names may compare equal if the base method is called.